wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -217,159 +146,181 @@ class PresentationActivity : FragmentActivity() {
-                                newTransferHelper(this)
-                            }
-                            stateDisplay.value = "Connected"
-                            Logger.i(TAG, "State: Connected")
-                        }
+                }
 
-                        State.REQUEST_AVAILABLE -> {
-                            stateDisplay.value = "Request Available"
-                            Logger.i(TAG, "State: Request Available")
-                            // start processing request and use processed request data to show consent prompt
-                            transferHelper?.startProcessingRequest(getDeviceRequest())
-                                ?.let { requestData ->
-                                    // update UI state object 'consentPromptData' so we can show ConsentPrompt
-                                    consentPromptData.value = ConsentPromptEntryFieldData(
-                                        credentialId = requestData.document.name,
-                                        documentName = requestData.document.documentConfiguration.displayName,
-                                        credentialData = requestData.document.documentConfiguration.mdocConfiguration!!.staticData,
-                                        documentRequest = requestData.documentRequest,
-                                        docType = requestData.docType,
-                                        verifier = requestData.trustPoint,
+                State.REQUEST_AVAILABLE -> {
+                    Logger.i(TAG, "State: Request Available")
+                    /**
+                     * Device request bytes have been transmitted via device retrieval helper listener,
+                     * Start the Presentment Flow where a Presentment is shown for every
+                     * [DocumentRequest] that has a suitable [Document].
+                     */
+                    lifecycleScope.launch {
+
+                        try {
+                            val deviceRequest = DeviceRequestParser(
+                                deviceRequestByteArray!!,
+                                deviceRetrievalHelper!!.sessionTranscript
+                            ).parse()
+
+                            deviceRequest.docRequests.forEach { docRequest ->
+                                // find a suitable Document for the docRequest
+                                val document = findSuitableDocumentForRequest(docRequest)
+                                // if document == null no suitable documents could be found then skip to next docRequest
+                                val mdocCredential = document.findCredential(
+                                    WalletApplication.CREDENTIAL_DOMAIN_MDOC,
+                                    Clock.System.now()
+                                ) as MdocCredential
+
+                                // extract the TrustPoint if possible
```

Comment:
This makes it sound like the `TrustPoint` is embedded in the request, which it isn't. I think it would be clearer if the comment said "See if we recognize the reader..."